backend/src/main/scala/cromwell/backend/wdl/OutputEvaluator.scala

```diff
@@ -1,31 +1,56 @@
 package cromwell.backend.wdl
 
+import cats.Applicative
+import cats.data.EitherT._
+import cats.data.{EitherT, NonEmptyList}
+import cats.instances.try_._
+import cats.instances.list._
+import cats.syntax.either._
+import cats.syntax.traverse._
 import cromwell.backend.BackendJobDescriptor
-import cromwell.core.JobOutput
-import lenthall.validation.ErrorOr.ShortCircuitingFlatMap
+import cromwell.core.{CallOutputs, JobOutput}
+import lenthall.validation.ErrorOr.ErrorOr
 import lenthall.validation.Validation._
-import wdl.LocallyQualifiedName
+import wdl.types.WdlType
 import wdl.values.WdlValue
-import wom.expression.IoFunctionSet
+import wom.callable.Callable.OutputDefinition
+import wom.expression.{IoFunctionSet, WomExpression}
 
-import scala.language.postfixOps
-import scala.util.{Success, Try}
+import scala.util.{Failure, Success, Try}
 object OutputEvaluator {
+  type OutputResult[A] = EitherT[Try, NonEmptyList[String], A]
```

Comment:
Because `evaluateValue` returns an `ErrorOr`, but can also throw an exception, and we want to handle both. I think it makes sense to have both because if whatever we use in `readFile` throws an exception for example, we want to keep it including the stacktrace (could be a GCS exception for example)

Reply:
It's the same concept as what @danbills did with CWL Parsing

Reply:
I'm thinking an ADT might make it more clear what's going on here rather than `Success(Left(...))` is one kind of failure and `Failure(..)` is a different kind of failure, but neither communicates to the reader what it means.

Reply:
The advantage of this is it can be `flatMap`ped easily. But I guess once it's been `flatMap`ped we could convert the result to an ADT